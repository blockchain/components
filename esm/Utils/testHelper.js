import React from 'react';
import { shallow, mount } from 'enzyme';
import { ThemeProvider } from 'styled-components';
import theme from '../Tools/theme';

function wrapWithTheme(fn, children, options) {
  var wrapper = fn(React.createElement(ThemeProvider, {
    theme: theme
  }, children), options);
  return wrapper[fn.name]({
    context: wrapper.instance().getChildContext()
  });
}

export function shallowWithTheme() {
  return wrapWithTheme.apply(void 0, [shallow].concat(Array.prototype.slice.call(arguments)));
}
export function mountWithTheme() {
  return wrapWithTheme.apply(void 0, [mount].concat(Array.prototype.slice.call(arguments)));
}