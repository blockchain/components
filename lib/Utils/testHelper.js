"use strict";

Object.defineProperty(exports, "__esModule", {
  value: true
});
exports.shallowWithTheme = shallowWithTheme;
exports.mountWithTheme = mountWithTheme;

var _react = _interopRequireDefault(require("react"));

var _enzyme = require("enzyme");

var _styledComponents = require("styled-components");

var _theme = _interopRequireDefault(require("../Tools/theme"));

function _interopRequireDefault(obj) { return obj && obj.__esModule ? obj : { default: obj }; }

function wrapWithTheme(fn, children, options) {
  var wrapper = fn(_react.default.createElement(_styledComponents.ThemeProvider, {
    theme: _theme.default
  }, children), options);
  return wrapper[fn.name]({
    context: wrapper.instance().getChildContext()
  });
}

function shallowWithTheme() {
  return wrapWithTheme.apply(void 0, [_enzyme.shallow].concat(Array.prototype.slice.call(arguments)));
}

function mountWithTheme() {
  return wrapWithTheme.apply(void 0, [_enzyme.mount].concat(Array.prototype.slice.call(arguments)));
}