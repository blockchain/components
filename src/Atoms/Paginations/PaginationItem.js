// @flow strict
import PropTypes from 'prop-types'
import * as React from 'react'
import styled, { type ReactComponentFunctional } from 'styled-components'

import injectE2E, { type E2ePropType } from '../../Tools/injectE2E'
import { fontSize, theme, type ThemePropType } from '../../Tools/interpolation'

const Wrapper: ReactComponentFunctional<E2ePropType> = styled.a.attrs({
  'data-e2e': injectE2E,
})`
  display: flex;
  flex-direction: row;
  justify-content: center;
  align-items: center;
  padding: 5px 8px;
  font-family: ${theme('fontPrimary')};
  font-size: ${fontSize('md')};
  font-weight: 500;
  color: ${(props: PropsType & ThemePropType) =>
    props.selected === true
      ? theme('paginationSelectedColor')(props)
      : theme('paginationColor')(props)};
  background-color: ${(props: PropsType & ThemePropType) =>
    props.selected === true
      ? theme('paginationSelectedBackgroundColor')(props)
      : theme('paginationBackgroundColor')(props)};
  border: 1px solid ${theme('paginationBorderColor')};
  border-radius: 5px;
  margin-right: 5px;
  cursor: pointer;

  &:hover {
    color: ${theme('paginationHoverColor')};
    background-color: ${(props: PropsType & ThemePropType) =>
      props.selected === true
        ? theme('paginationSelectedBackgroundColor')(props)
        : theme('paginationHoverBackgroundColor')(props)};
  }
`

type PropsType = {
  +children?: React.Node,
  +e2e?: string,
  +selected?: boolean,
}

const PaginationItem = ({ children, ...rest }: PropsType) => (
  <Wrapper {...rest}>{children}</Wrapper>
)

PaginationItem.propTypes = {
  children: PropTypes.node,
  e2e: PropTypes.string.isRequired,
  selected: PropTypes.bool,
}

PaginationItem.defaultProps = {
  e2e: 'defaultPaginationItem',
  selected: false,
}

export default PaginationItem
