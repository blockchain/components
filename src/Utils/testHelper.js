import React from 'react'
import { shallow, mount } from 'enzyme'
import { ThemeProvider } from 'styled-components'
import theme from '../Tools/theme'

function wrapWithTheme(fn, children, options) {
  const wrapper = fn(
    <ThemeProvider theme={theme}>{children}</ThemeProvider>,
    options,
  )

  return wrapper[fn.name]({
    context: wrapper.instance().getChildContext(),
  })
}

export function shallowWithTheme() {
  return wrapWithTheme(shallow, ...arguments)
}

export function mountWithTheme() {
  return wrapWithTheme(mount, ...arguments)
}
