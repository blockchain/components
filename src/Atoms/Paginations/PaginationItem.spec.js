import React from 'react'
import { mountWithTheme } from '../../Utils/testHelper'

import PaginationItem from './PaginationItem'

describe('PaginationItem', () => {
  it('should render correctly', () => {
    const tree = mountWithTheme(<PaginationItem selected />)

    expect(tree).toMatchSnapshot()
  })
})
