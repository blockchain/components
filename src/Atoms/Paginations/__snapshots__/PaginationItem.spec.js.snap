// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`PaginationItem should render correctly 1`] = `
.c0 {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-flex-direction: row;
  -ms-flex-direction: row;
  flex-direction: row;
  -webkit-box-pack: center;
  -webkit-justify-content: center;
  -ms-flex-pack: center;
  justify-content: center;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
  padding: 5px 8px;
  font-size: 1rem;
  font-weight: 500;
  color: #FFF;
  background-color: #10ADE4;
  border: 1px solid #EEF2F5;
  border-radius: 5px;
  margin-right: 5px;
  cursor: pointer;
}

.c0:hover {
  color: #383838;
  background-color: #10ADE4;
}

<ThemeProvider
  theme={
    Object {
      "buttonBackgroundColor": "#004A7C",
      "buttonColor": "#FFF",
      "buttonHoverColor": "#0068af",
      "cartridgeBackgroundColor": "#10ADE4",
      "cartridgeColor": "#FFF",
      "circleBadgeBackgroundColor": "#CCC",
      "circleBadgeHoverColor": "#004A7C",
      "dropdownArrowColor": "#545456",
      "dropdownColor": "#545456",
      "fontPrimary": "'IBM Plex Sans', 'Helvetica Neue', Arial, sans-serif;",
      "footerCategoryColor": "#040d25",
      "iconColor": "#545456",
      "iconHoverColor": "#757679",
      "inputBackgroundColor": "#FFF",
      "inputBorderColor": "#CCC",
      "inputColor": "#545456",
      "inputPlaceholderColor": "#999B9E",
      "linkColor": "#545456",
      "linkHoverColor": "#004A7C",
      "navbarDropdownContentBackgroundColor": "#FFF",
      "navbarDropdownContentBorderColor": "#999B9E",
      "navbarDropdownContentShadowColor": "rgba(0,0,0,0.25)",
      "navbarDropdownContentShadowColor2": "rgba(0,0,0,0.10)",
      "navbarDropdownDescriptionColor": "#999B9E",
      "navbarDropdownHeaderColor": "#FFF",
      "navbarDropdownTitleColor": "#3558A8",
      "navbarDropdownTitleMobileColor": "#FFF",
      "navbarMenuBagroundColor": "#004A7C",
      "navbarTogglerColor": "#FFF",
      "paginationBackgroundColor": "rgba(0, 0, 0, 0)",
      "paginationBorderColor": "#EEF2F5",
      "paginationColor": "#383838",
      "paginationHoverBackgroundColor": "rgba(16,173,228,0.1)",
      "paginationHoverColor": "#383838",
      "paginationSelectedBackgroundColor": "#10ADE4",
      "paginationSelectedColor": "#FFF",
      "palette": Object {
        "azure": "#3558A8",
        "bigStone": "#152343",
        "biscay": "#153A62",
        "black": "#000",
        "blackAlpha05": "rgba(0,0,0,0.05)",
        "blackAlpha10": "rgba(0,0,0,0.10)",
        "blackAlpha25": "rgba(0,0,0,0.25)",
        "blackAlpha75": "rgba(0,0,0,0.75)",
        "blueWood": "#27324B",
        "cerulean": "#10ADE4",
        "ceruleanAlpha10": "rgba(16,173,228,0.1)",
        "gold": "#FFCA00",
        "gray0": "#F8F8F8",
        "gray1": "#EEF2F5",
        "gray2": "#CCC",
        "gray3": "#999B9E",
        "gray4": "#757679",
        "gray5": "#545456",
        "gray6": "#383838",
        "gray7": "#040d25",
        "orient": "#004A7C",
        "powderBlue": "#B2D5E5",
        "transparent": "rgba(0, 0, 0, 0)",
        "white": "#FFF",
        "whiteAlpha10": "rgba(255,255,255,0.1)",
        "whiteAlpha75": "rgba(255,255,255,0.75)",
      },
      "panelBackgroundColor": "#FFF",
      "panelShadowColor": "#EEF2F5",
      "searchBarIconBackgroundColor": "#FFCA00",
      "searchBarInputBackgroundColor": "#FFF",
      "searchBarUnderlinedBackgroundColor": "rgba(0, 0, 0, 0)",
      "searchBarUnderlinedBorderColor": "#CCC",
      "sectionBackgroundColor": "rgba(0, 0, 0, 0)",
      "tabMenuBarColor": "#FFCA00",
      "tabMenuBorderColor": "#EEF2F5",
      "tableCellEvenBackgroundColor": "#F8F8F8",
      "tableCellOddBackgroundColor": "#FFF",
      "tableShadowColor": "#EEF2F5",
      "textColor": "#545456",
      "togglerBackgroundColor": "#FFF",
      "togglerBorderColor": "#EEF2F5",
      "togglerColor": "#10ADE4",
      "togglerSelectedBackgroundColor": "#10ADE4",
      "togglerSelectedColor": "#FFF",
      "tooltipBackgroundColor": "#383838",
      "tooltipBorderColor": "#383838",
      "tooltipColor": "#FFF",
    }
  }
>
  <PaginationItem
    selected={true}
  >
    <styled.a
      selected={true}
    >
      <a
        className="c0"
        selected={true}
      />
    </styled.a>
  </PaginationItem>
</ThemeProvider>
`;
